Jenkinsfile

```diff
@@ -78,15 +72,37 @@ pipeline {
         echo """
 BUILD_TIMESTAMP=${env.BUILD_TIMESTAMP}
 BUILD_TYPE=${env.BUILD_TYPE}
-TARGET_PLATFORM=${env.TARGET_PLATFORM}
 ECLIPSE_SIGN=${env.ECLIPSE_SIGN}
 PROMOTE=${env.PROMOTE}
 BRANCH_NAME=${env.BRANCH_NAME}
 """
       }
     }
 
-    stage('Build') {
+    stage('Test With Latest') {
+      steps {
+        wrap([$class: 'Xvnc', useXauthority: true]) {
+          dir('.') {
+            sh '''
+              mvn \
+                --fail-at-end \
+                --no-transfer-progress \
+                --update-snapshots \
+                -P!promote \
+                -Dmaven.repo.local=xpect-local-maven-repository \
+                -Dmaven.artifact.threads=16 \
+                -Declipsesign=false \
+                -Dtycho-version=${TYCHO_VERSION} \
+                -Dtarget-platform=${TARGET_PLATFORM_LATEST} \
+                clean \
+                integration-test
+              '''
+          }
+        }
+      }
+    }
+
+    stage('Build With Primary') {
```

Comment:
do we need a clean -fdx or something the like between the steps?

Reply:
(in the old Jenkijnsfile we used a separate checkout location)

Reply:
I don't think so.  Both do mvn clean.  Neither do an install so the maven cache can be reused.